### PR TITLE
Updates for tests

### PR DESCRIPTION
**Pull request by ngodfraind**
It needs to be merged before #125 

#125 will have to be able to handle 2 bundles.ini files (one for testing, one for prod) or some issues may arise. However I don't think it's very important to do it now. 

Note: the test suite is rather sluggish with all the api tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -48,7 +48,7 @@ before_script:
   - npm run bower
   - composer build
   - php app/console doctrine:database:create --env=test
```
Comment (stefk) [inline .travis.yml]: This line can go too.

**Comment by stefk** [comment]
> #125 will have to be able to handle 2 bundles.ini files (one for testing, one for prod) or some issues may arise.

Can you explain?

**Comment by ngodfraind** [comment]
If someone disable some plugins, they'll also be disabled for the test
suite and it won't run properly (if the disabled plugins has some tests).
This is not very important.

2016-04-18 15:50 GMT+02:00 Stéphane Klein <notifications@github.com>:

> #125 <https://github.com/claroline/Distribution/pull/125> will have to be
> able to handle 2 bundles.ini files (one for testing, one for prod) or some
> issues may arise.
>
> Can you explain?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/claroline/Distribution/pull/160#issuecomment-211389093>
>



-- 
Godfraind Nicolas
5, rue de la Carrière
6592 Monceau-Imbrechies
GSM: 0472 94 80 99

**Comment by stefk** [comment]
I see, maybe just making the bundle handler ignore the active flag in test environment would be easier.

(still the frontend path issue)

**Comment by ngodfraind** [comment]
Looks good.

LGTM

**Comment by stefk** [comment]
LGTM

**Comment by stefk** [comment]
@ngodfraind, squash?

**Comment by ngodfraind** [comment]
Yes.
Btw, any tips for the cli squash in order to not screw up everything ?

**Comment by stefk** [comment]
The best I can do for now is:

    git rebase -i -p HEAD~x

where x is the number of commits I want to squash. The `-p` option is for "preserve merges". Note that even with those options, if there are some merges in the middle, you'll see all the associated commits in the interactive rebase interface. If you let them on "pick" that's not a problem. You can squash/reorder your own commits as you like though.

There might be a better way but I don't know it yet.

**Comment by stefk** [comment]
OMG this just made the test suite 5x slower (from 1 min to 5 min)... There're obviously big problems in the way those new tests are set up.

**Comment by ngodfraind** [comment]
This is what I said.

2016-04-18 17:26 GMT+02:00 Stéphane Klein <notifications@github.com>:

> OMG this just made the test suite 5x slower (from 1 min to 5 min)...
> There're obviously big problems in the way those new tests are set up.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/claroline/Distribution/pull/160#issuecomment-211430945>
>



-- 
Godfraind Nicolas
5, rue de la Carrière
6592 Monceau-Imbrechies
GSM: 0472 94 80 99

**Comment by stefk** [comment]
You said "rather sluggish", but we are way beyond that. This should be fixed asap.

**Comment by ngodfraind** [comment]
This is what happens with functionnal tests. You can check them in the
Tests/API folder. It's mainly group and user controller. However, they do
not take 4 min to be run outside of the test suite.

2016-04-18 17:31 GMT+02:00 Stéphane Klein <notifications@github.com>:

> You said "rather sluggish", but we are way beyond that. This should be
> fixed asap.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/claroline/Distribution/pull/160#issuecomment-211433016>
>



-- 
Godfraind Nicolas
5, rue de la Carrière
6592 Monceau-Imbrechies
GSM: 0472 94 80 99

**Comment by ngodfraind** [comment]
the --process-isolation could help (with and update of the persister wich
shouldn't create everything related to users).

2016-04-18 17:34 GMT+02:00 Godfraind Nicolas <nicolas.godfraind@gmail.com>:

> This is what happens with functionnal tests. You can check them in the
> Tests/API folder. It's mainly group and user controller. However, they do
> not take 4 min to be run outside of the test suite.
>
> 2016-04-18 17:31 GMT+02:00 Stéphane Klein <notifications@github.com>:
>
>> You said "rather sluggish", but we are way beyond that. This should be
>> fixed asap.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/claroline/Distribution/pull/160#issuecomment-211433016>
>>
>
>
>
> --
> Godfraind Nicolas
> 5, rue de la Carrière
> 6592 Monceau-Imbrechies
> GSM: 0472 94 80 99
>



-- 
Godfraind Nicolas
5, rue de la Carrière
6592 Monceau-Imbrechies
GSM: 0472 94 80 99

**Comment by ngodfraind** [comment]
I fix the widget and I change the tests for the User section.

2016-04-18 17:35 GMT+02:00 Godfraind Nicolas <nicolas.godfraind@gmail.com>:

> the --process-isolation could help (with and update of the persister wich
> shouldn't create everything related to users).
>
> 2016-04-18 17:34 GMT+02:00 Godfraind Nicolas <nicolas.godfraind@gmail.com>
> :
>
>> This is what happens with functionnal tests. You can check them in the
>> Tests/API folder. It's mainly group and user controller. However, they do
>> not take 4 min to be run outside of the test suite.
>>
>> 2016-04-18 17:31 GMT+02:00 Stéphane Klein <notifications@github.com>:
>>
>>> You said "rather sluggish", but we are way beyond that. This should be
>>> fixed asap.
>>>
>>> —
>>> You are receiving this because you were mentioned.
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/claroline/Distribution/pull/160#issuecomment-211433016>
>>>
>>
>>
>>
>> --
>> Godfraind Nicolas
>> 5, rue de la Carrière
>> 6592 Monceau-Imbrechies
>> GSM: 0472 94 80 99
>>
>
>
>
> --
> Godfraind Nicolas
> 5, rue de la Carrière
> 6592 Monceau-Imbrechies
> GSM: 0472 94 80 99
>



-- 
Godfraind Nicolas
5, rue de la Carrière
6592 Monceau-Imbrechies
GSM: 0472 94 80 99

**Pull request merged**